docs/source-2.0/tutorials/full-stack-tutorial.rst

```diff
@@ -0,0 +1,880 @@
+
+.. tip:: 
+    This tutorial does not assume you are an expert in Smithy, but you may find it helpful to work through the
+    :doc:`../quickstart` before beginning this tutorial.
+
+Setting up the project
+======================
+This application will consist of 4 major components:
+
+1. A model, which defines the service
+2. A server, which handles requests for coffee 
+3. A client, which is generated from the model
+4. A web application, which uses the client to make requests to the server
+
+--------------
+Pre-requisites
+--------------
+To follow this tutorial, you will need to install a few tools:
+
+* :doc:`Smithy CLI <../guides/smithy-cli/cli_installation>`
+* `Node.js (>= 16) <https://nodejs.org/en/download>`_ and `yarn <https://yarnpkg.com/getting-started/install>`_
+
+------
+Set up
+------
+Once you have the prerequisites installed, the Smithy CLI has a useful command to easily bootstrap projects from
+`the repository of example smithy projects <https://github.com/smithy-lang/smithy-examples>`_. Execute the following
+command to set up the initial project:
+
+.. code-block:: sh
+    :caption: ``/bin/sh``
+
+    smithy init -t full-stack-application
+
+After running this command, you should have the project in the ``full-stack-application`` directory.
+The directory tree should look like:
+
+.. code-block:: sh
+    :caption: ``/bin/sh``
+
+    full-stack-application
+    ├── Makefile
+    ├── README.md
+    ├── app
+    │   ├── ...
+    ├── client
+    │   ├── ...
+    ├── server
+    │   ├── ...
+    └── smithy
+        ├── ...
+
+The ``README.md`` file contains important information about the project, like its directory structure and how to
+build or run it. However, for this tutorial, we will show you which commands to run, when to run them, and
+the expected outputs.
+
+Please run the following command to set the project up, so you can follow this tutorial:
+
+.. code-block:: sh
+    :caption: ``/bin/sh``
+
+    make init
+
+.. hint:: If you get stuck and want to completely start over, run ``make reset && make init`` .
+
+    Also, the complete project code for this tutorial is available
+    `here <https://github.com/smithy-lang/smithy-examples/tree/main/tutorials/full-stack-application>`_.
+
+Modeling the service
+====================
+With the basic framework for the project established, let's walk through how to model our coffee service.
+The service should provide a few capabilities:
+
+* provide a menu of coffees
+* provide the ability to order a coffee
+* provide the ability to check the status of an order
+
+------------------
+Adding the service
+------------------
+The service shape is the entry-point of our API, and is where we define the operations our service exposes to a
+consumer. First and foremost, let's define the initial service shape without any operations. Open the ``main.smithy``
+file and add the following:
+
+.. important:: For code blocks, the name of the current file is given in the top-left corner.
```

Comment:
For the file names, and when running various commands (i.e. `smithy build`), maybe we should have the path be relative to the project root, so for the code block below, it would be `smithy/model/main.smithy`. And for commands running in the `smithy` directory, maybe `cd smithy/ && smithy build` or something.

Reply:
I'll update the paths to be relative to where their "current working directory" should be at that point. Adding `cd`'s probably isn't a good idea.